helion/runtime/ref_mode.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+import enum
+import os
+import threading
+from typing import TYPE_CHECKING
+from typing import cast
+
+import torch
+from torch.overrides import BaseTorchFunctionMode
+
+from helion._compiler.compile_environment import CompileEnvironment
+from helion._compiler.compile_environment import tls as ce_tls
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .settings import Settings
+
+_thread_local = threading.local()
+
+
+class RefMode(enum.Enum):
+    """Reference mode for kernel execution."""
+
+    OFF = "off"
+    EAGER = "eager"
+
+
+def _get_ref_mode_from_env() -> RefMode:
+    """Get reference mode from environment variables."""
+    # Check for environment variables
+    ref_eager = os.environ.get("HELION_REF_EAGER", "").lower() in ("1", "true", "yes")
```

Comment:
Maybe HELION_INTERPRET to match triton?  Or something forward looking to having multiple modes?